modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```
@@ -454,41 +488,61 @@ else if (Objects.equals(BinaryExpression.Operation.OR, operation)) {
 		}
 		else {
 			ObjectField objectField = _objectFieldLocalService.fetchObjectField(
-				_objectDefinitionId, String.valueOf(left));
+				objectDefinitionId, String.valueOf(left));
 
 			if ((objectField != null) &&
 				StringUtil.equals(
 					objectField.getBusinessType(),
 					ObjectFieldConstants.BUSINESS_TYPE_MULTISELECT_PICKLIST)) {
 
-				predicate = _contains(left, right);
+				predicate = _contains(left, right, objectDefinitionId);
 			}
 		}
 
 		if (predicate != null) {
 			return predicate;
 		}
 
-		if (_objectRelationship != null) {
+		return _getExpressionPredicate(
+			_getColumn(left, objectDefinitionId), operation,
+			_getValue(left, objectDefinitionId, right));
+	}
+
+	private Predicate _getPredicateForRelationships(
+		Object left,
+		UnsafeBiFunction<String, Long, Predicate, Exception>
+			getRelatedObjectFieldPredicateUnsafeBiFunction) {
+
+		String leftString = (String)left;
+
+		String[] complexPropertyChunks = leftString.split(StringPool.SLASH);
+
+		String relationshipName = complexPropertyChunks[0];
+
+		ObjectRelationship objectRelationship = _fetchObjectRelationship(
+			relationshipName);
+
+		if (objectRelationship != null) {
+			String objectFieldName = complexPropertyChunks[1];
+
 			try {
-				return _getPredicateForRelationships(operation, left, right);
+				return _getPredicateForRelationships(
+					objectRelationship,
+					getRelatedObjectFieldPredicateUnsafeBiFunction.apply(
+						objectFieldName,
+						_getRelatedObjectDefinitionId(
+							_objectDefinitionId, objectRelationship)));
 			}
 			catch (Exception exception) {
-				if (_log.isDebugEnabled()) {
-					_log.debug(exception);
-				}
-
-				return null;
+				throw new RuntimeException(exception);
 			}
 		}
 
-		return _getExpressionPredicate(
-			_getColumn(left, _objectDefinitionId), operation,
-			_getValue(left, _objectDefinitionId, right));
+		return null;
```

I think in this case, this should throws an exception instead of returning null